inject/src/main/java/io/micronaut/inject/BeanDefinition.java

```diff
@@ -253,6 +254,30 @@ default <R> Optional<ExecutableMethod<T, R>> findMethod(String name, Class<?>...
         return Optional.empty();
     }
 
+    /**
+     * Whether an {@link ExecutableMethod} exists which is annotated with the supplied annotation
+     *
+     * @param methodName          The method name
+     * @param argumentTypes The argument types
+     * @return Whether an {@link ExecutableMethod} exists which is annotated with the supplied annotation
+     * @since 4.3.0
+     */
+    default boolean existsMethodWithAnnotation(Class<? extends Annotation> annotationClass, String methodName, Class<?>... argumentTypes) {
+        return findMethod(methodName, argumentTypes).map(method -> method.hasAnnotation(annotationClass)).orElse(false);
+    }
+
+    /**
+     * Whether an {@link ExecutableMethod} exists which is annotated with {@link Blocking}.
+     *
+     * @param methodName          The method name
+     * @param argumentTypes The argument types
+     * @return Whether an {@link ExecutableMethod} exists which is annotated with with {@link Blocking}.
+     * @since 4.3.0
+     */
+    default boolean existsMethodAnnotatedWithBlocking(String methodName, Class<?>... argumentTypes) {
```

Comment:
I don't think this method adds any value, and we shouldn't have methods for every annotation type here. Just remove it and you can pass `Blocking.class` when calling the above method

Reply:
resolved via https://github.com/micronaut-projects/micronaut-core/pull/10261/commits/b02ad851014ab4a1a9d15cabb3acba13c3d1f114